docs/datamodel/build-a-datamodel.rst

```diff
@@ -0,0 +1,627 @@
+        marshmallow_loader
+    from ..marshmallow import MetadataSchemaV1
+
+    json_v1 = marshmallow_loader(MetadataSchemaV1)
+
+Note, that you are not required to use Marshmallow for deserialization, but it
+allows you to use advanced data validation rules on your REST API.
+
+Define templates
+----------------
+In order to display records not only on your REST API, but also provide
+search interface and landing pages for your record you need to provide
+templates that render your records.
+
+You will need two different types of templates:
+
+- Search result template
+- Landing page template
+
+The templates are stored in two different folders (``static`` and
+``templates``):
+
+.. code-block:: shell
+
+    |-- my_datamodel
+    |   |-- static
+    |   |   `-- templates
+    |   |       `-- my_datamodel
+    |   |           `-- results.html
+    |   |-- templates
+    |   |   `-- my_datamodel
+    |   |       `-- record.html
+
+
+**Search result template**
+
+The Invenio search interface is run by a JavaScript application, and thus the
+template is rendered client side in the user's browser. The template uses data
+received by the REST API and thus your REST API must be able to deliver all
+information you would like to render in the template.
+
+**Landing page template**
+
+The landing page for a single record is rendered on the server-side using a
+Jinja template.
+
+Configure the UI
+----------------
+Last step after having defined all the different schemas, serializers, loaders
+and templates is to configure your REST API and landing pages for your records.
+
+This is all done from the ``config.py``:
+
+.. code-block:: shell
+
+    |-- my_datamodel
+    |   |-- config.py
+
+**Landing page**
+
+Let's start by configuring the landing page:
+
+.. code-block:: python
+
+    RECORDS_UI_ENDPOINTS = {
+        'recid': {
+            'pid_type': 'recid',
+            'route': '/records/<pid_value>',
+            'template': 'my_datamodel/record.html',
+        },
+    }
+
+Here an explanation of the different keys:
+
+* ``pid_type``: Defines the persistent identifier type which the resolver
+  should use to lookup records. Invenio provides an internal persistent
+  identifier type called ``recid`` which is an auto-incrementing integer.
+* ``route``: URL endpoint under which to expose the landing pages.
+* ``template``: Template to use when rendering the landnge page.
+* ``recid``: Unique name of the endpoint. If this is the primary landing page,
+  it must be named the same as the value of ``pid_type`` (i.e. ``recid``).
+
+Configure the REST API
+----------------------
+Configuring the REST API is done similarly to the landing pages via the
+``RECORDS_REST_ENDPOINTS`` configuration variable in ``config.py``:
+
+**Persistent identifier type**
+
+First you provide the persistent identifier type used by the resolver. You also
+need to configure a persistent identifier minter and fetcher. In the scaffolded
```

Comment:
Links to [minters](https://invenio-pidstore.readthedocs.io/en/latest/usage.html#minters) and [fetchers](https://invenio-pidstore.readthedocs.io/en/latest/usage.html#fetchers) documentation?